app/Http/Controllers/ChurchController.php

```diff
@@ -23,12 +24,20 @@ public function __construct()
     public function index()
     {
         $this->authorize('viewAny', Church::class);
-        $churches = Church::all();
+        $user = auth()->user();
+        $religions = Religion::all();
+        
+        if ($user->is_admin){
+          $churches = Church::all();
+        } else {
+          $churches = Church::where(['religion_id' => $user->religion->id])->get();
+        }
+        
         $addresses = [];
```

Comment:
you could only pass in the religion or for a non-admin user you could pass only their religion.